tools/pythonpkg/tests/fast/arrow/test_filter_pushdown.py

```diff
@@ -137,314 +167,465 @@ def test_filter_pushdown_numeric(self, duckdb_cursor):
+        assert duckdb_cursor.execute("SELECT count(*) from arrow_table where a IS NULL").fetchone()[0] == 1
         # Try Is Not Null
-        assert duckdb_conn.execute("SELECT count(*) from testarrow where a IS NOT NULL").fetchone()[0] == 3
+        assert duckdb_cursor.execute("SELECT count(*) from arrow_table where a IS NOT NULL").fetchone()[0] == 3
 
         # Try And
         assert (
-            duckdb_conn.execute(
-                "SELECT count(*) from testarrow where a='2010-01-01 10:00:01' and b ='2008-01-01 00:00:01'"
+            duckdb_cursor.execute(
+                "SELECT count(*) from arrow_table where a = '2010-01-01 10:00:01' and b = '2008-01-01 00:00:01'"
             ).fetchone()[0]
             == 0
         )
         assert (
-            duckdb_conn.execute(
-                "SELECT count(*) from testarrow where a ='2020-03-01 10:00:01' and b = '2010-01-01 10:00:01' and c = '2020-03-01 10:00:01'"
+            duckdb_cursor.execute(
+                "SELECT count(*) from arrow_table where a = '2020-03-01 10:00:01' and b = '2010-01-01 10:00:01' and c = '2020-03-01 10:00:01'"
             ).fetchone()[0]
             == 1
         )
         # Try Or
         assert (
-            duckdb_conn.execute(
-                "SELECT count(*) from testarrow where a = '2020-03-01 10:00:01' or b ='2008-01-01 00:00:01'"
+            duckdb_cursor.execute(
+                "SELECT count(*) from arrow_table where a = '2020-03-01 10:00:01' or b ='2008-01-01 00:00:01'"
             ).fetchone()[0]
             == 2
         )
 
-    def test_filter_pushdown_date(self, duckdb_cursor):
-        duckdb_conn.execute("CREATE TABLE test_date (a  DATE, b DATE, c DATE)")
-        duckdb_conn.execute(
-            "INSERT INTO  test_date VALUES ('2000-01-01','2000-01-01','2000-01-01'),('2000-10-01','2000-10-01','2000-10-01'),('2010-01-01','2000-10-01','2010-01-01'),(NULL,NULL,NULL)"
+    @pytest.mark.parametrize('create_table', [create_pyarrow_pandas, create_pyarrow_table])
+    @pytest.mark.parametrize(
+        ['data_type', 'value'],
+        [
+            ['TINYINT', 127],
+            ['SMALLINT', 32767],
+            ['INTEGER', 2147483647],
+            ['BIGINT', 9223372036854775807],
+            ['UTINYINT', 255],
+            ['USMALLINT', 65535],
+            ['UINTEGER', 4294967295],
+            ['UBIGINT', 18446744073709551615],
+        ],
+    )
+    def test_filter_pushdown_integers(self, duckdb_cursor, data_type, value, create_table):
+        duckdb_cursor.execute(
+            f"""
+            CREATE TABLE tbl as select {value}::{data_type} as i
+        """
+        )
+        expected = duckdb_cursor.table('tbl').fetchall()
+        filter = "i > 0"
+        rel = duckdb_cursor.table('tbl')
+        arrow_table = create_table(rel)
+        actual = duckdb_cursor.sql(f"select * from arrow_table where {filter}").fetchall()
+        assert expected == actual
```

Comment:
Does this make sense? And should we expect it to work?

```suggestion
        actual = duckdb_cursor.sql(f"select * from arrow_table where {filter}").fetchall()
        assert expected == actual

        # Test with equivalent prepared statement
        actual = duckdb_cursor.execute("select * from arrow_table where i > ?", (0,)).fetchall()
        assert expected == actual

        # Test equality
        actual = duckdb_cursor.execute("select * from arrow_table where i = ?", (value,)).fetchall()
        assert expected == actual
```

Reply:
Added it